src/main/java/com/salesforce/storm/spout/dynamic/coordinator/SpoutCoordinator.java

```diff
@@ -208,6 +208,19 @@ public synchronized boolean hasVirtualSpout(final VirtualSpoutIdentifier virtual
         return runningSpouts.containsKey(virtualSpoutIdentifier);
     }
 
+    /**
+     * Get a {@link DelegateSpout} instance from the {@link SpoutCoordinator}.
+     *
+     * This is useful is you want to manipulate the filter chain or alter the ending state after a {@link DelegateSpout} has
+     * been added to the {@link SpoutCoordinator}.
+     *
+     * @param virtualSpoutIdentifier identifier for teh {@link DelegateSpout} instance to get from the {@link SpoutCoordinator}.
+     * @return {@link DelegateSpout} instance
+     */
+    public synchronized DelegateSpout getVirtualSpout(final VirtualSpoutIdentifier virtualSpoutIdentifier) {
+        return runningSpouts.get(virtualSpoutIdentifier).getSpoutRunner().getSpout();
```

Comment:
Probably need to handle the case where you ask for a non-existing virtual spout.  Right now I'm guessing it would NPE.